enterprise/cmd/repo-updater/internal/authz/perms_syncer_worker_test.go

```diff
@@ -96,11 +94,119 @@ func TestPermsSyncerWorker(t *testing.T) {
 	observationCtx := &observation.TestContext
 	dummySyncer := &dummySyncerWithErrors{
 		repoIDErrors: map[api.RepoID]struct{}{2: {}},
+	}
+
+	workerStore := MakeStore(observationCtx, db.Handle(), SyncTypeRepo)
+	worker := MakeTestWorker(ctx, observationCtx, workerStore, dummySyncer, SyncTypeRepo)
+	go worker.Start()
+	t.Cleanup(func() { worker.Stop() })
```

Comment:
```suggestion
	t.Cleanup(worker.Stop)
```

first-class functions, baby!